tests/integration/multicluster/reachability.go

```diff
@@ -24,49 +24,55 @@ import (
 	"istio.io/istio/pkg/test/framework/components/namespace"
 	"istio.io/istio/pkg/test/framework/components/pilot"
 	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
 )
 
+const mcReachabilitySvcPerCluster = 3
+
 // ReachabilityTest tests that services in 2 different clusters can talk to each other.
-func ReachabilityTest(t *testing.T, pilots []pilot.Instance) {
+func ReachabilityTest(t *testing.T, ns namespace.Instance, pilots []pilot.Instance) {
 	framework.NewTest(t).
 		Label(label.Multicluster).
 		Run(func(ctx framework.TestContext) {
 			ctx.NewSubTest("reachability").
 				Run(func(ctx framework.TestContext) {
-					ns := namespace.NewOrFail(ctx, ctx, namespace.Config{
-						Prefix: "mc-reachability",
-						Inject: true,
-					})
-
 					// Deploy services in different clusters.
-					// There are more in the remote cluster to tease out cases where remote proxies inconsistently
-					// use different discovery servers.
-					var a, b, c, d, e, f echo.Instance
-					echoboot.NewBuilderOrFail(ctx, ctx).
-						With(&a, newEchoConfig("a", ns, ctx.Environment().Clusters()[0], pilots)).
-						With(&b, newEchoConfig("b", ns, ctx.Environment().Clusters()[1], pilots)).
-						With(&c, newEchoConfig("c", ns, ctx.Environment().Clusters()[1], pilots)).
-						With(&d, newEchoConfig("d", ns, ctx.Environment().Clusters()[1], pilots)).
-						With(&e, newEchoConfig("e", ns, ctx.Environment().Clusters()[1], pilots)).
-						With(&f, newEchoConfig("f", ns, ctx.Environment().Clusters()[1], pilots)).
-						BuildOrFail(ctx)
+					// There are multiple instances in each cluster to tease out cases where remote proxies inconsistently
```

Comment:
OK, I think I understand why you're doing this now. It's really more about the "source" service in each cluster. Before the cluster-local fix, each source service would potentially get a different discovery server, so some would work and some wouldn't. Is that right?

Maybe we can expand on this comment to make it more clear?